### PR TITLE
add replit code

### PR DESCRIPTION
**Pull request by abacaj**
Added replit code. Majority of the changes were taken from the original impl found on ggml. Should add, my c++ is rusty/and not as experienced with it. Have it working here: https://github.com/abacaj/replit-3B-inference

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 setup(
     name=name,
-    version="0.2.10",
+    version="0.2.11",
```
Comment (marella) [inline setup.py:24]: This change is not required. I will update this before releasing a new version.

File: models/llms/replit.cc [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+  bool Eval(const std::vector<gpt_vocab::id> &tokens, const int threads,
+            const int n_past) override
+  {
+    return replit_eval(model_, threads, n_past, tokens, logits_,
+                       mem_per_token_);
+  }
+
+private:
+  replit_model model_;
+  mutable std::string current_word_;
```
Comment (marella) [inline models/llms/replit.cc:725]: Nit: You can call it something like `detokenized_text_` to make it easier to understand that it is used by detokenize method.

**Review by marella — CHANGES_REQUESTED** [review]
Thanks. Do you have a link to download model that I can use for testing.

**Comment by abacaj** [comment]
> Thanks. Do you have a link to download model that I can use for testing.

Hi yes, model is hosted here: https://huggingface.co/abacaj/Replit-v2-CodeInstruct-3B-ggml. Will make the other changes you mentioned

File: models/llms/replit.cc [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+    if (context_length > 0)
+    {
+      model_.hparams.n_ctx = context_length;
+    }
+    if (!replit_model_load(filename, model_, replit_tokenizer_))
+    {
+      return false;
+    }
+    n_ctx_ = model_.hparams.n_ctx;
+    return true;
```
[the remark this answers is not in the document]
Comment (marella) [reply models/llms/replit.cc:713]: You can now remove the `IsEosToken`, `EosToken`, `VocabSize`, `VocabSize`, `Sample` methods as they are now same as the base methods.

**Review by marella — CHANGES_REQUESTED** [review]

**Comment by marella** [comment]
> > Thanks. Do you have a link to download model that I can use for testing.
> 
> Hi yes, model is hosted here: https://huggingface.co/abacaj/Replit-v2-CodeInstruct-3B-ggml. Will make the other changes you mentioned

Thanks, I will test this tomorrow.

**Review by marella — APPROVED** [review]

**Pull request merged**